molgenis-core-ui/src/main/resources/js/component/Form.js

```diff
@@ -49,14 +51,44 @@
 			};
 		},
 		componentWillReceiveProps : function(nextProps) {
-			var newState = {
-				errorMessages : {},
-				validate: false,
-				showModal: true,
-				submitMsg: null
-			};
+			var entity = this.props.entity;
+			var entityInstance = this.props.entityInstance;
+			var resetProps = false;
```

Comment:
Heh, I meant for the resetProps variable, not for newState variable. Cause it's the state that gets reset, not the props. Anyways, what do you think?